### ISSUE TITLE
Root folders missing from File Browser

### ISSUE THREAD

**Issue by haydnjonesdesign** [libretro/RetroArch#10580]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description
On linux file browser doesn't recognise user created directories at /
For example /test/dir

### Expected behavior
File browser lists all dir's in / 
For example
/bin
/boot
/dev
**/test**

### Actual behavior
Dir's in / are not shown in the file browser

### Steps to reproduce the bug

1. Install Retroarch via snap. 
`sudo snap install retroarch`
2. Create a dir. 
`sudo mkdir /test`
3. Launch retroarch
4. Settings > Directories
5. Edit directory and select /

Also tried 
- running retroarch as sudo. 
- checked chown on dir which is set to root:root
- uninstall/reinstall
- running as another user
- scanned code base looking for anything obvious that would limit the display of dir's
- searched past issues + forums
all with no luck.

### Bisect Results
Fresh install of Retroarch on Debian via SnapCraft.io. 

### Version/Commit
- RetroArch: 1.8.6

### Environment information
- OS: Debian Buster
- Compiler: N.A

**Comment by i30817** [comment]
You maybe screwed up the permissions of the folder and made it only readable to sudo.

**Comment by i30817** [comment]
I just tried to sudo mkdir /test and it appears to me. Might it be because you're using a container or something like that?

Anyway don't ever run retroarch as sudo, unless you want 'fun' bugs involving the owner of files changing. Be warned that a least your playlists and configs changed owner right now.

**Comment by hizzlekizzle** [comment]
Since you mentioned that you're using the snap package, I suspect this is related to snap permissions/sandboxing.

**Comment by haydnjonesdesign** [comment]
I too believe it's a snap issue. I was unaware how snap actually worked and it looks like it creates a container for each snap install with the developer determining which interfaces are available to the snap (i.e. network, removable-media, system-folder, home) . I don't believe this is related to super user. 

The snap install installs in strict mode which does limit the system folders. If the developer supplies a classic install then snap can install it with `snap install retroarch --classic` which may resolve the issue however retroarch does not supply this. The flag --devmode may resolve this but some snap benefits are lost.

The rationale for using snap was Debian Buster fails to install due to failed dependencies. 

The use case of using system folders for snap installs is probably low and may be resolved if the developer supplies a classic install or compiling retroarch. Eventually I'll try compile but I hate to think of the issues Debian will though up thanks to its stable packages.

**Comment by ivanleoncz** [comment]
Agree. Seems a snap issue. Something similar with VLC from Snap also happens, not allowing Snap to find a folder from a mounted directory over a secondary storage device.

**Comment by gouchi** [comment]
Hi,

Will it possible to open this issue to the [dedicated repository](https://github.com/libretro/retroarch-snap) ?

Thank you.

Kind regards,
